sapi/fpm/fpm/fpm_main.c

```diff
@@ -1918,6 +1920,8 @@ consult the installation file that came with this distribution, or visit \n\
 
 			php_execute_script(&file_handle);
 
+			zend_destroy_file_handle(&file_handle);
```

Comment:
I might be missing something but from the code this seems to me like a serious leak that this is fixing. If I read it correctly, the filename and file fd is leaked on every request. We might need to do some testing to confirm it as I can't believe this could be happening for so long without noticing.

Reply:
I didn't use a tool or anything to find this. I just stumbled upon this and saw there was no freeing. Do you have suggestions on how to test it reliably?

Reply:
Running FPM with 1 pool with static pm and one child and the hit it with mulitple requests. Just GDB (requires setting `set follow-fork-mode child`) might show if it's getting freed or maybe trying to use Valgrind or other tools might show it as well. I had some issues with Valgrind in past when trying to debug child but it was for callgrind. Memcheck might work though.

Reply:
I don't see an increase in entries in `/proc/<pid of FPM>/`. I guess this is due to the file handle using a stream instead of a FILE*, but I'm not sure to how it all works internally and I didn't look that far into it tbh.

But I do see a memory leak report even without using Valgrind, *but only when using opcache* starting from the second request to the same file. The script name is being leaked (at least). Possibly other resources (that aren't being reported?) could be leaked as well, idk for sure. When I apply this PR I no longer see the leak.

That one leak looks like:
```
[Sun Mar 12 00:54:15 2023]  Script:  '-'
/home/niels/php-src/Zend/zend_string.h(150) :  Freeing 0x00007fbaca65f4e0 (64 bytes), script=-
```